django/db/models/expressions.py

```diff
@@ -927,6 +931,77 @@ def convert_value(self, value, expression, connection, context):
         return value
 
 
+class SubQuery(Expression):
+    """
+    Notes about both SubQuery() and Exists().
+
+    MyModel.objects.annotate(
+        field=SubQuery(RelatedModel.objects.filter(date=Today())
+                                           .filter(parent=F('outerq__relation'))
+                                           .values('other_field')[:1]),
+        found=Exists(OtherModel.objects.filter(thing=True, parent=F('outerq__relation')))
+    )
+
+    Note that for SubQuery, unless you are aggregating the sub query, you'll want to use
+    a [:1] on the subquery to use a LIMIT 1. The Exists() is not subject to this. You'll probably
+    also want to use a .values() with a single argument, unless you are using something that
+    will deal with multiple fields (like building a JSON object, maybe).
+
+    What about SubqueryConstraint? Can we leverage that? Is there any point?
+    """
+    template = '(%(subquery)s)'
+
+    def __init__(self, subquery, output_field=None, **extra):
+        self.subquery = subquery
+        self.extra = extra
+        super(SubQuery, self).__init__(output_field)
+
+    def resolve_expression(self, query=None, allow_joins=True, reuse=None, summarize=False, for_save=False):
+        clone = self.copy()
+        clone.is_summary = summarize
+
+        def resolve(child):
+            if hasattr(child, 'children'):
+                [resolve(_child) for _child in child.children]
+            if hasattr(child, 'rhs') and isinstance(child.rhs, F):
+                fieldname = child.rhs.name.split('__')[1]
+                if fieldname == 'pk':
```

Comment:
You can likely just resolve the F() expression using resolve_expression() once you have removed the outerq prefix from the F obj.